armada/api/middleware.py

```diff
@@ -40,49 +38,67 @@
 
 class AuthMiddleware(object):
 
-    def process_request(self, req, resp):
-        req_context = req.context
-        token = req.get_header('X-Auth-Token')
-
-        # Validate token
-        #self._session = self._validate_token(token)
-	self._session = self._password_auth()	
-	
-	try:
-	    roles = self._session.get(url='http://192.168.148.91/v3/role_assignments')	
-	except:
-	    raise falcon.HTTPUnauthorized('Authentication required',
-                                         ('Authentication was not successful.'))	
-   	#req_context.add_roles(self._get_roles())
-	print(self._get_roles())
- 
-    def _validate_token(self, token):
-        auth = v3.Token(auth_url="http://192.168.148.91/v3", token=token)
-	return session.Session(auth=auth)
-
-    def _get_roles(self):
-        try:
-	    roles_resp = self._session.get(url="http://192.168.148.91/v3/role_assignments")
-	except:
-	    raise falcon.HTTPUnauthorized('Authentication required',
-                                         ('Authentication was not successful.'))	 
-	roles = roles_resp.json()['role_assignments'] #json.loads(roles_resp.text)[0][roles]
-	
-        role_list = []
-        for role in roles:
-            role_list.append(role['id'])
-	print(role_list)	
-        return role_list
-
-    def _password_auth(self):
-	auth = v3.Password(auth_url="http://192.168.148.91:35357/v3",
+	def process_request(self, req, resp):
+		token = req.get_header('X-Auth-Token')
+
+        # Validate token and get user session
+        #self._session = self._get_token_session(token)
+
+		# TODO: Remove method call (testing purposes)
+		self._session = self._password_auth()
+
+		# Add token roles to context
+		req.context['roles'] = self._get_roles()
+
+	def _get_roles(self):
+		request_url = CONF.auth_url + '/role_assignments'
+
+		# Get roles associated with user
+		try:
+			resp = self._session.get(request_url)
+		except:
+			raise falcon.HTTPUnauthorized('Authentication required',
+	                              ('Authentication token is invalid.'))
+								  
+		json_resp = resp.json()['role_assignments']
+
+		roles = []
```

Comment:
This can also use a list comprehension:
`roles = [r['role']['id'].encode('utf-8') for r in json_resp]`

Also consider adding this to the try/except block to catch any typing errors.